public/app/features/alerting/unified/components/rules/central-state-history/EventListSceneObject.tsx

```diff
@@ -31,86 +17,84 @@ import {
 } from 'app/types/unified-alerting-dto';
 
 import { stateHistoryApi } from '../../../api/stateHistoryApi';
+import { labelsMatchMatchers, parseMatchers } from '../../../utils/alertmanager';
 import { GRAFANA_RULES_SOURCE_NAME } from '../../../utils/datasource';
 import { stringifyErrorLike } from '../../../utils/misc';
 import { hashLabelsOrAnnotations } from '../../../utils/rule-id';
 import { AlertLabels } from '../../AlertLabels';
 import { CollapseToggle } from '../../CollapseToggle';
 import { LogRecord } from '../state-history/common';
-import { useRuleHistoryRecords } from '../state-history/useRuleHistoryRecords';
-
-const LIMIT_EVENTS = 250;
-
-const HistoryEventsList = ({ timeRange }: { timeRange?: TimeRange }) => {
-  const styles = useStyles2(getStyles);
-
-  // Filter state
-  const [eventsFilter, setEventsFilter] = useState('');
-  // form for filter fields
-  const { register, handleSubmit, reset } = useForm({ defaultValues: { query: '' } }); //  form for search field
+import { isLine, isNumbers } from '../state-history/useRuleHistoryRecords';
+
+import { LABELS_FILTER } from './CentralAlertHistoryScene';
+
+export const LIMIT_EVENTS = 5000; // limit is hard-capped at 5000 at the BE level.
+
+/**
+ *
+ * This component displays a list of history events.
+ * It fetches the events from the history api and displays them in a list.
+ * The list is filtered by the labels in the filter variable and by the time range variable in the scene graph.
+ */
+export const HistoryEventsList = ({
+  timeRange,
+  valueInfilterTextBox,
+}: {
+  timeRange?: TimeRange;
+  valueInfilterTextBox: VariableValue;
+}) => {
   const from = timeRange?.from.unix();
   const to = timeRange?.to.unix();
-  const onFilterCleared = useCallback(() => {
-    setEventsFilter('');
-    reset();
-  }, [setEventsFilter, reset]);
 
   const {
     data: stateHistory,
     isLoading,
     isError,
     error,
-  } = stateHistoryApi.endpoints.getRuleHistory.useQuery(
-    {
-      from: from,
-      to: to,
-      limit: LIMIT_EVENTS,
-    },
-    {
-      refetchOnFocus: true,
-      refetchOnReconnect: true,
-    }
+  } = stateHistoryApi.endpoints.getRuleHistory.useQuery({
+    from: from,
+    to: to,
+    limit: LIMIT_EVENTS,
+  });
+
+  const { historyRecords: historyRecordsNotSorted } = useRuleHistoryRecords(
+    stateHistory,
+    valueInfilterTextBox.toString()
   );
 
-  const { historyRecords } = useRuleHistoryRecords(stateHistory, eventsFilter);
+  const historyRecords = historyRecordsNotSorted.sort((a, b) => b.timestamp - a.timestamp);
```

Comment:
You mentioned in standup that you had to sort this because otherwise something in the scenes/graph didn't work - could we leave a comment to say why? (I can't remember exactly what the reason was 😁 )

Reply:
nono, my comment in the standup was about the part3 where I'm building another time series graph (and I already added an explanation in the code 😎). In here we are sorting because as you mentioned when I was doing the demo, the results were not sorted.